combat/CombatSystem.cpp

```diff
@@ -1266,12 +1310,38 @@ namespace {
 
 
             // select target object
-            int target_idx = RandInt(0, valid_target_ids.size() - 1);
-            int target_id = *std::next(valid_target_ids.begin(), target_idx);
+            std::shared_ptr<UniverseObject> target;
+            DebugLogger() << "Target precision: " << std::to_string(weapon.precision);
+            for (int t = 1; t < weapon.precision ; t++) {
+                // Do all but the last round of finding a valid target; the simple last round is separate for performance reasons
+                DebugLogger() << "Find target - Round: " << std::to_string(t) << " (try to lock it if it matches preferredPrey conditions)";
+                int target_idx = RandInt(0, valid_target_ids.size() - 1);
+                int target_id = *std::next(valid_target_ids.begin(), target_idx);
+
+                target = combat_state.combat_info.objects.Object(target_id);
+                if (!target) {
+                    ErrorLogger() << "AutoResolveCombat couldn't get target object with id " << target_id;
+                    continue;
+                }
 
-            auto target = combat_state.combat_info.objects.Object(target_id);
+                if (weapon.AimsFor(target)) {
+                    // shooting
+                    break;
+                }
+                target = nullptr; // signal that last round should be executed
+            }
+            if (!target) {
+                DebugLogger() << "Find target - Round: " << std::to_string(weapon.precision) << " (Final round ignores preferredPrey condition)";
```

Comment:
comment this block?

Reply:
i add something for the final round if that is the block you mean